### PR TITLE
Migrate platform reference to Upbound official providers

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* Migrate platform reference to https://marketplace.upbound.io/providers/upbound/provider-aws
* Remove outdated UI annotations
* Extend postgres XRD with passwordRef

Concerns:
* Readme
* Packaging

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
k apply -f examples/cluster.yaml
k apply -f examples/postgres-claim.yaml
...
k get managed
Warning: Please use v1beta1 version of SNS group.
NAME                                                      CHART                   VERSION   SYNCED   READY   STATE   REVISION   DESCRIPTION   AGE
release.helm.crossplane.io/platform-ref-aws-8phhf-fd2ck   kube-prometheus-stack   34.5.1    False                                             68m

NAME                                                              READY   SYNCED   EXTERNAL-NAME           AGE
internetgateway.ec2.aws.upbound.io/platform-ref-aws-8phhf-rtrc5   True    True     igw-07d51b4ce6889cbc1   68m

NAME                                                         READY   SYNCED   EXTERNAL-NAME           AGE
routetable.ec2.aws.upbound.io/platform-ref-aws-8phhf-n9qdm   True    True     rtb-0611cf381448d7537   68m

NAME                                                                        READY   SYNCED   EXTERNAL-NAME                AGE
mainroutetableassociation.ec2.aws.upbound.io/platform-ref-aws-8phhf-l28qx   True    True     rtbassoc-0b5be2ed8de843749   68m

NAME                                                    READY   SYNCED   EXTERNAL-NAME                       AGE
route.ec2.aws.upbound.io/platform-ref-aws-8phhf-862sr   True    True     r-rtb-0611cf381448d75371080289494   68m

NAME                                                                    READY   SYNCED   EXTERNAL-NAME                AGE
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-8phhf-g7fv9   True    True     rtbassoc-0bcd3704bbdf1bfa9   68m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-8phhf-jn2jg   True    True     rtbassoc-098605487600c7d67   68m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-8phhf-4nrcv   True    True     rtbassoc-001e5aaff23c9f611   68m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-8phhf-fbps5   True    True     rtbassoc-033eabb564c9e8527   68m

NAME                                                            READY   SYNCED   EXTERNAL-NAME          AGE
securitygroup.ec2.aws.upbound.io/platform-ref-aws-8phhf-b6456   True    True     sg-0eb4d103c9a31a795   68m

NAME                                                  READY   SYNCED   EXTERNAL-NAME           AGE
vpc.ec2.aws.upbound.io/platform-ref-aws-8phhf-klxth   True    True     vpc-054be240b1952b3dd   68m
vpc.ec2.aws.upbound.io/platform-ref-aws-5lwn2-8s5cv   False   True     vpc-02b924cd1d3c99bc4   3h29m

NAME                                                     READY   SYNCED   EXTERNAL-NAME              AGE
subnet.ec2.aws.upbound.io/platform-ref-aws-8phhf-jb67n   True    True     subnet-0384a47f3a53b6bbf   68m
subnet.ec2.aws.upbound.io/platform-ref-aws-8phhf-wcj5g   True    True     subnet-0136f57ea77520f63   68m
subnet.ec2.aws.upbound.io/platform-ref-aws-8phhf-d42bf   True    True     subnet-06bbb600b17dd75e7   68m
subnet.ec2.aws.upbound.io/platform-ref-aws-8phhf-bxd5b   True    True     subnet-09bb454ed188e864e   68m

NAME                                                                READY   SYNCED   EXTERNAL-NAME       AGE
securitygrouprule.ec2.aws.upbound.io/platform-ref-aws-8phhf-25k77   True    True     sgrule-4258495705   68m

NAME                                                        READY   SYNCED   EXTERNAL-NAME                  AGE
nodegroup.eks.aws.upbound.io/platform-ref-aws-8phhf-kvrw2   True    True     platform-ref-aws-8phhf-kvrw2   68m

NAME                                                      READY   SYNCED   EXTERNAL-NAME                  AGE
cluster.eks.aws.upbound.io/platform-ref-aws-8phhf-r4tsc   True    True     platform-ref-aws-8phhf-r4tsc   68m

NAME                                                   READY   SYNCED   EXTERNAL-NAME                  AGE
role.iam.aws.upbound.io/platform-ref-aws-8phhf-822s9   True    True     platform-ref-aws-8phhf-822s9   68m
role.iam.aws.upbound.io/platform-ref-aws-8phhf-ql6nn   True    True     platform-ref-aws-8phhf-ql6nn   68m

NAME                                                                   READY   SYNCED   EXTERNAL-NAME                                             AGE
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-8phhf-82gvl   True    True     platform-ref-aws-8phhf-ql6nn-20220913222859982900000091   68m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-8phhf-xhqvt   True    True     platform-ref-aws-8phhf-ql6nn-20220913222901874900000092   68m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-8phhf-m57qd   True    True     platform-ref-aws-8phhf-822s9-20220913222937496100000093   68m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-8phhf-j4k9p   True    True     platform-ref-aws-8phhf-ql6nn-20220913222939213900000094   68m

NAME                                                          READY   SYNCED   EXTERNAL-NAME                     AGE
instance.rds.aws.upbound.io/platform-ref-aws-db-5n885-wtlgv   True    True     platform-ref-aws-db-5n885-wtlgv   16m

NAME                                                             READY   SYNCED   EXTERNAL-NAME                     AGE
subnetgroup.rds.aws.upbound.io/platform-ref-aws-db-5n885-8xt7t   True    True     platform-ref-aws-db-5n885-8xt7t   16m
```

### CODE REVIEW

File: database/postgres/composition.yaml [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
             dbSubnetGroupNameSelector:
               matchControllerRef: true
-            dbInstanceClass: db.t2.small
-            masterUsername: masteruser
+            instanceClass: db.t3.micro
+            username: masteruser
             engine: postgres
-            engineVersion: "9.6"
+            engineVersion: "13.7"
             skipFinalSnapshotBeforeDeletion: true
```
Comment (stevendborrelli) [inline database/postgres/composition.yaml:37]: Getting a   `destroy failed: final_snapshot_identifier is required when skip_final_snapshot is false:`, when trying to delete.

**Review by stevendborrelli — CHANGES_REQUESTED** [review]
Unfortunately, I am unable to fully test the EKS cluster due to the following error: 

```
 apply failed: error creating Security Group (platform-ref-aws-cluster): InvalidGroup.Duplicate: The security group 'platform-ref-aws-cluster' already exists for VPC 'vpc-013520bb25d84e361'
```

Changes needed: 
- update providerConfig group to upbound.io
- look at deletion policy for postgresDB

**Comment by ytsarev** [comment]
@stevendborrelli thanks a lot for the test! I've updated ProviderConfigs

Just to clarify:
* Does it fail for you immediately in the creation state?
* `look at deletion policy for postgresDB` - could you elaborate on what is required here? Do you mean `deletionPolicy: Orphan` by default?

**Comment by stevendborrelli** [comment]
@ytsarev Yes, it looks like the VPC I am testing in already has that security group, so it fails to create.

**Comment by ytsarev** [comment]
@stevendborrelli i've addressed both issues, thanks again for the review. I suggest we retest this platform with the latest provider-aws release tomorrow

File: README.md [hunk 2]
```diff
@@ -159,35 +159,8 @@ kubectl get pkg
...
-kubectl apply -f examples/aws-default-provider.yaml
-```
-The output will look like:
-
-```shell
-provider.aws.crossplane.io/default create
-```
-Crossplane resources use the ProviderConfig named ```default``` if no specific ProviderConfig is specified, so this ProviderConfig will be the default for all AWS resources.
+Refer to [official marketplace
+documentation](https://marketplace.upbound.io/providers/upbound/provider-aws/v0.12.0/docs/configuration)
```
Comment (stevendborrelli) [inline README.md:163]: Is there a way to point to the most recent version?

**Review by stevendborrelli — CHANGES_REQUESTED** [review]
Everything looks good, was able to generate every resource except the Helm deployment.

Regarding the helm issue, apparently the kubeconfig is not being generated properly for the generated `ProviderConfig`: 

```
  Warning  CannotConnectToProvider  7s (x4 over 22m)   managed/release.helm.crossplane.io  cannot create new rest config using provider secret: currentContext not set in kubeconfig
```

File: cluster/eks/composition.yaml [hunk 3]
```diff
@@ -37,18 +37,18 @@ spec:
...
             policyArn: arn:aws:iam::aws:policy/AmazonEKSClusterPolicy
-            roleNameSelector:
+            roleSelector:
               matchControllerRef: true
               matchLabels:
                 role: controlplane
       name: clusterRolePolicyAttachment
     - base:
-        apiVersion: eks.aws.crossplane.io/v1beta1
+        apiVersion: eks.aws.upbound.io/v1beta1
```
Comment (muvaf) [inline cluster/eks/composition.yaml:51]: ClusterAuth is ready to be used to get the kubeconfig https://marketplace.upbound.io/providers/upbound/provider-aws/v0.14.0/resources/eks.aws.upbound.io/ClusterAuth/v1beta1

**Comment by ytsarev** [comment]
```
 k get managed
NAME                                                      CHART                   VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-aws-xvhmz-wmd4m   kube-prometheus-stack   34.5.1    True     True    deployed   1          Install complete   46m

NAME                                                  READY   SYNCED   EXTERNAL-NAME           AGE
vpc.ec2.aws.upbound.io/platform-ref-aws-xvhmz-hkz76   True    True     vpc-006cb9cd6f169d76e   46m

NAME                                                                        READY   SYNCED   EXTERNAL-NAME                AGE
mainroutetableassociation.ec2.aws.upbound.io/platform-ref-aws-xvhmz-qts8l   True    True     rtbassoc-01aa96b3305f64e21   46m

NAME                                                                    READY   SYNCED   EXTERNAL-NAME                AGE
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-xvhmz-hr2rf   True    True     rtbassoc-00c2ffd2ce5a4cb95   46m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-xvhmz-d92d2   True    True     rtbassoc-08b05d7beed4bf586   46m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-xvhmz-fsldm   True    True     rtbassoc-0804a91f9230a6567   46m
routetableassociation.ec2.aws.upbound.io/platform-ref-aws-xvhmz-tfmrl   True    True     rtbassoc-03aa13507cd31fdc0   46m

NAME                                                            READY   SYNCED   EXTERNAL-NAME          AGE
securitygroup.ec2.aws.upbound.io/platform-ref-aws-xvhmz-8vg6m   True    True     sg-0bec64463b8402d38   46m

NAME                                                                READY   SYNCED   EXTERNAL-NAME       AGE
securitygrouprule.ec2.aws.upbound.io/platform-ref-aws-xvhmz-x9msb   True    True     sgrule-2213752351   46m

NAME                                                         READY   SYNCED   EXTERNAL-NAME           AGE
routetable.ec2.aws.upbound.io/platform-ref-aws-xvhmz-lr8zj   True    True     rtb-09722427098e9637e   46m

NAME                                                    READY   SYNCED   EXTERNAL-NAME                       AGE
route.ec2.aws.upbound.io/platform-ref-aws-xvhmz-7m45h   True    True     r-rtb-09722427098e9637e1080289494   46m

NAME                                                     READY   SYNCED   EXTERNAL-NAME              AGE
subnet.ec2.aws.upbound.io/platform-ref-aws-xvhmz-bjtms   True    True     subnet-0feb73c9165bc41b9   46m
subnet.ec2.aws.upbound.io/platform-ref-aws-xvhmz-v6pj5   True    True     subnet-0abdd5098d8e41ce1   46m
subnet.ec2.aws.upbound.io/platform-ref-aws-xvhmz-vxv5w   True    True     subnet-02e6590762b467f0e   46m
subnet.ec2.aws.upbound.io/platform-ref-aws-xvhmz-7npb8   True    True     subnet-06b57d24ac902fe77   46m

NAME                                                              READY   SYNCED   EXTERNAL-NAME           AGE
internetgateway.ec2.aws.upbound.io/platform-ref-aws-xvhmz-lvm7r   True    True     igw-094f72ebab582e68c   46m

NAME                                                      READY   SYNCED   EXTERNAL-NAME                  AGE
cluster.eks.aws.upbound.io/platform-ref-aws-xvhmz-z9qc4   True    True     platform-ref-aws-xvhmz-z9qc4   46m

NAME                                                          READY   SYNCED   EXTERNAL-NAME                  AGE
clusterauth.eks.aws.upbound.io/platform-ref-aws-xvhmz-4t2xl   True    True     platform-ref-aws-xvhmz-4t2xl   46m

NAME                                                        READY   SYNCED   EXTERNAL-NAME                  AGE
nodegroup.eks.aws.upbound.io/platform-ref-aws-xvhmz-6hdch   True    True     platform-ref-aws-xvhmz-6hdch   46m

NAME                                                                   READY   SYNCED   EXTERNAL-NAME                                             AGE
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-xvhmz-hflt6   True    True     platform-ref-aws-xvhmz-k74d9-20220921222626803900000001   46m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-xvhmz-m7kpv   True    True     platform-ref-aws-xvhmz-k74d9-20220921222629247300000001   46m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-xvhmz-549qt   True    True     platform-ref-aws-xvhmz-k74d9-20220921222632178800000001   46m
rolepolicyattachment.iam.aws.upbound.io/platform-ref-aws-xvhmz-tq7k8   True    True     platform-ref-aws-xvhmz-grhqk-20220921222656328200000001   46m

NAME                                                   READY   SYNCED   EXTERNAL-NAME                  AGE
role.iam.aws.upbound.io/platform-ref-aws-xvhmz-k74d9   True    True     platform-ref-aws-xvhmz-k74d9   46m
role.iam.aws.upbound.io/platform-ref-aws-xvhmz-grhqk   True    True     platform-ref-aws-xvhmz-grhqk   46m

NAME                                                                    READY   SYNCED   EXTERNAL-NAME                                                                                                  AGE
openidconnectprovider.iam.aws.upbound.io/platform-ref-aws-xvhmz-rs8jw   True    True     arn:aws:iam::609897127049:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/FACE87BF6BB004CC11704F649B220CFE   16m
```

```
k get -f examples/cluster-claim.yaml
NAME               READY   CONNECTION-SECRET             AGE
platform-ref-aws   True    platform-ref-aws-kubeconfig   47m
```

Looks all green with the new `ClusterAuth`.

Last thing for this iteration is to rework `subnetIds` for reliable deletion.

**Comment by ytsarev** [comment]
I've just pushed the latest changes (thanks @muvaf for the id-through-status propagation changes!)

It looks all good for both creation and deletion

@stevendborrelli  could you please retest it? the current state is also packaged at  xpkg.upbound.io/upbound/platform-ref-aws-staging:v0.4.2 for convenience. Thanks!

**Comment by stevendborrelli** [comment]
Notes from 10/22 Meeting: https://docs.google.com/document/d/1BQT0uCvEfeRa0ygja8Y8M9cEPt9Aq2j6caBfUrBSQpU/edit#

**Comment by stevendborrelli** [comment]
More of an upstream issue, but after deleting the cluster claim, the helm resource is left stranded:

```
kubectl get managed
NAME                                                      CHART                   VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-aws-xj2qm-z7bz5   kube-prometheus-stack   34.5.1    False    False   deployed   1          Install complete   71m
```

```
Status:
  At Provider:
    Release Description:  Install complete
    Revision:             1
    State:                deployed
  Conditions:
    Last Transition Time:  2022-09-22T16:25:31Z
    Message:               connect failed: failed to extract kubeconfig: cannot get credentials secret: Secret "505ee2e5-19e3-4e3f-b253-9b2e7e30874a-ekscluster" not found
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
    Last Transition Time:  2022-09-22T16:24:31Z
    Reason:                Deleting
    Status:                False
    Type:                  Ready
  Synced:                  true
Events:
  Type     Reason                   Age                   From                                Message
  ----     ------                   ----                  ----                                -------
  Normal   CreatedExternalResource  36m                   managed/release.helm.crossplane.io  Successfully requested creation of external resource
  Normal   DeletedExternalResource  23m                   managed/release.helm.crossplane.io  Successfully requested deletion of external resource
  Warning  CannotConnectToProvider  2m32s (x22 over 69m)  managed/release.helm.crossplane.io  failed to extract kubeconfig: cannot get credentials secret: Secret "505ee2e5-19e3-4e3f-b253-9b2e7e30874a-ekscluster" not found
```

**Review by muvaf — APPROVED** [review]
Great! Let's go!

File: crossplane.yaml [hunk 4]
```diff
@@ -32,64 +32,11 @@ metadata:
...
 spec:
   crossplane:
-    version: ">=v1.2.0-0"
+    version: ">=v1.9.0-0"
```
Comment (jbw976) [inline crossplane.yaml:37]: I think this says the minimum version of Crossplane that this package will install on.  Does this package require functionality in crossplane v1.9+? If not, then I think we shouldn't move past the currently supported versions of Crossplane, which still include v1.7.

File: crossplane.yaml [hunk 5]
```diff
@@ -32,64 +32,11 @@ metadata:
...
...LjU5MjQ1ODQuOTc1ODg2Ni0xLjAyOTc0OSAxLjM0MTg0NC0uNDM3MjkwNy4zODEyMDU3LS45NTkyMTgyLjY1NTY3MzgtMS41NjU3ODI3Ljg1MzkwMDctLjYzNDc3NjcuMjEzNDc1Mi0xLjI5Nzc2NTguMzIwMjEyOC0yLjAxNzE3OTQuMzIwMjEyOHoiLz48cGF0aCBmaWxsPSIjRkY5OTAwIiBkPSJNNDkuODc1NTA3IDQxLjAwNTkzNzJjLTQuNzA1MTM2NyAzLjY3NDQwNTEtMTEuNTQxMTcxMiA1LjYyNTAxNTItMTcuNDE4ODczOCA1LjYyNTAxNTItOC4yMzc1NjQ2IDAtMTUuNjU5OTUzNjEtMy4yMjA3NzQ4LTIxLjI2NjA3Mzk4LTguNTczNjExOS0uNDQzNDg0MTYtLjQyMzM4ODItLjA0MjkwMzk4LS45OTc5ODY1LjQ4NjI0NTE0LS42NjUzMjQ0IDYuMDYzNzYyODUgMy43MTk3NjgyIDEzLjU0MzIxNDE0IDUuOTcyNzk4NCAyMS4yODAzNzUyNCA1Ljk3Mjc5ODQgNS4yMTk5ODQ2IDAgMTAuOTU0ODE2OS0xLjE0OTE5NjYgMTYuMjMyMDA2Ny0zLjUwODA3NC43ODY1NzMtLjM3ODAyNTIgMS40NTg3MzU0LjU0NDM1NjMuNjg2MzIwNyAxLjE0OTE5Njd6bTEuOTU5NDI0OS0yLjM1ODg3NzNjLS42MDA3OTg4LS44MTY1MzQ1LTMuOTc1NzY5LS4zOTMxNDYzLTUuNTA2MDExMS0uMTk2NTczMi0uNDU3Nzg1NS4wNjA0ODQxLS41MjkyOTIxLS4zNjI5MDQyLS4xMTQ0MTA2LS42ODA0NDUzIDIuNjg4NjQ5Ni0xLjk5NTk3MzIgNy4xMDc2MTY3LTEuNDIxMzc0OCA3LjYyMjYwNzYtLjc1NjA1MDUuNTE0ODQ3Ny42ODA0NDU0LS4xNDMwMTMzIDUuMzUyODM3MS0yLjY2MDE5IDcuNTkwNzQ2My0uMzg1OTkyOC4zNDc3ODMyLS43NTc4MjczLjE2NjMzMTEtLjU4NjM1NDQtLjI4NzI5OTEuNTcyMDUzMS0xLjQ5Njk3OTkgMS44NDUwMTQzLTQuODY4OTY0OCAxLjI0NDM1ODUtNS42NzAzNzgyeiIvPjwvZz48L3N2Zz4=
```
Comment (jbw976) [inline crossplane.yaml:87]: what icon will be used for this package now?

Comment (ytsarev) [reply crossplane.yaml:87]: @jbw976 I have consulted with @hasheddan today and he confirmed that:
* we are good to remove all annotations that are not part of the xpkg spec at https://crossplane.io/docs/v1.9/reference/xpkg.html#object-annotations
* icon annotation is not used at the moment and we manually inject icons for some specific packages
@hasheddan please confirm that my understanding is correct

**Review by jbw976 — APPROVED** [review]
thanks for the quick follow up yury!

**Pull request merged**